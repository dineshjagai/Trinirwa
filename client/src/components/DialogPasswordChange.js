/* eslint-disable no-unused-vars */
import React, { useContext, useState } from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import isValidPassword from './authentication';
import { updatePassword } from './Module';
import idContext from './Context';

export default function DialogPasswordChange() {
  const username = useContext(idContext);
  const [open, setOpen] = useState(false);
  const [checker, setChecker] = useState(true);
  const [isValid, setIsValid] = useState(true);
  const handleClickOpen = () => {
    setChecker(true);
    setOpen(true);
    setIsValid(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleCloseChange = () => {
    const oldPassword = document.getElementById('oldPassword').value;
    const newPassword = document.getElementById('newPassword').value;
    const isNotEqual = oldPassword !== newPassword;
    const oldIsNull = oldPassword !== '';
    const newIsNull = newPassword !== '';
    if (checker && isValid && isNotEqual && newIsNull && oldIsNull) {
      updatePassword(username, newPassword, oldPassword).then((response) => {
        // if (response.status === 401) {

        // } else {

        // }
      }).catch((err) => console.log(err.message));
    }
  };
  return (
    <div>
      <button id="change" style={{ color: '#00695c', border: 'none', backgroundColor: 'inherit' }} type="button" onClick={handleClickOpen}>Change password</button>
      <Dialog open={open} onClose={handleClose} aria-labelledby="form-dialog-title">
        <DialogTitle id="form-dialog-title">Change password</DialogTitle>

        <DialogContent>
          <TextField
            autoFocus
            margin="dense"
            id="oldPassword"
            label="Old password"
            type="password"
            fullWidth
          />

          <TextField
            error={!isValid}
            autoFocus
            onChange={(event) => {
              setIsValid(isValidPassword(event.target.value));
            }}
            margin="dense"
            id="newPassword"
            label="Input new password"
            type="password"
            fullWidth
            helperText={isValid ? 'Password is strong!' : 'Password needs a min length of 8. It must consist of at 8 ...'}
          />

          <TextField
            error={!checker}
            onChange={(event) => {
              setChecker(event.target.value === document.getElementById('newPassword').value);
            }}
            autoFocus
            margin="dense"
            id="newPasswordConfirmed"
            label="Confirm new password"
            type="password"
            fullWidth
            helperText={checker ? '' : 'New password must be identical'}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="#00695c">
            Cancel
          </Button>
          <Button onClick={handleCloseChange} color="primary">
            Update
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}
