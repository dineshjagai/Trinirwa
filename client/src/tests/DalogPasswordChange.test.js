import React from 'react';
import renderer from 'react-test-renderer';
import DialogPasswordChange from '../components/DialogPasswordChange';

test('Test DialogPasswordChange snapshot', () => {
  const handle = (cid) => {
    if (cid) {
      console.log("owner");
    } else {
      console.log("HERE {id}");
    }
  };
  const component = renderer.create(<DialogPasswordChange />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
