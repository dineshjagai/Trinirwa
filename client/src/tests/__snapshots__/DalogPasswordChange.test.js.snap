// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DialogPasswordChange snapshot 1`] = `
<div>
  <button
    id="change"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "border": "none",
        "color": "#00695c",
      }
    }
    type="button"
  >
    Change password
  </button>
</div>
`;
